.github/workflows/deploy-release.yml

```diff
@@ -0,0 +1,24 @@
+name: Deploy Release 
```

Comment:
This workflow and the former one are doing the same task: build wheels and release.
Why do you need to split?

If it is only because not always you want to publish on PyPI, put an input flag on the former one. In this way, you also avoid the extra burden of downloading artifacts (you can still upload, if you want to be able to check them anyhow)

Reply:
And of course you can make the `PYPI_TOKEN` optional, in case the deployment is skipped.